src/odemis/acq/feature.py

```diff
@@ -119,3 +143,322 @@ def load_project_data(path: str) -> dict:
+        return expected_movement_time
+
+    def estimate_total_time(self) -> float:
+        """Estimate the total time for the acquisition task."""
+        n_active_features = sum(f.status.value != FEATURE_DEACTIVE for f in self.features)
+        expected_movement_time = self.estimate_movement_time()
+        expected_acq_time = self.estimate_acquisition_time()
+        total_time = expected_acq_time * n_active_features + expected_movement_time
+        logging.debug(f"Estimated total time {total_time}s for {n_active_features} features")
+        return total_time
+
+    def run(self) -> Tuple[List[model.DataArray], Exception]:
+        """Run the acquisition task."""
+        exp = None
+        self._future._task_state = RUNNING
+        self._future.set_progress(end=time.time() + self.estimate_total_time() + 2)
+        try:
+            with self._future._task_lock:
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+
+                # move to FM IMAGING posture
+                current_posture = self.pm.getCurrentPostureLabel()
+                logging.debug(f"Current Posture: {POSITION_NAMES[current_posture]}")
+
+                if current_posture != FM_IMAGING:
+                    raise ValueError(f"Currently only supported when at {POSITION_NAMES[FM_IMAGING]} posture")
+                    # TODO: enable this when the cryo switch is fixed
+                    logging.debug(
+                        f"Moving to {POSITION_NAMES[FM_IMAGING]} from {POSITION_NAMES[current_posture]}"
+                    )
+                    f = self.pm.cryoSwitchSamplePosition(FM_IMAGING)
+                    f.result()
+
+            for feature in self.features:
+                logging.debug(f"starting acquisition task for {feature.name.value}")
+
+                if feature.status.value == FEATURE_DEACTIVE:
+                    logging.info(f"Skipping feature {feature.name.value} because it is deactivated")
+                    continue
+
+                # move to feature position
+                self._move_to_site(feature)
+
+                # acquire data
+                self._acquire_feature(feature)
+
+        except CancelledError:
+            logging.debug("Stopping because acquisition was cancelled")
+            raise
+        except Exception as exp:
+            logging.exception(f"The acquisition failed: {exp}")
+            raise
+        finally:
+            self._future._task_state = FINISHED
+
+        return [], exp
+
+    def _export_data(self, feature: CryoFeature, data: List[model.DataArray]) -> None:
+        """
+        Called to export the acquired data.
+        data: the returned data/images from the future
+        """
+
+        filename = add_feature_info_to_filename(feature, self.filename)
+
+        self.exporter.export(filename, data)
+        logging.info("Acquisition saved as file '%s'.", filename)
+
+
+def acquire_at_features(
```

Comment:
Please add a test case. Just call this function once, with a couple of features, on a METEOR simulator.
No need to check too much. Something like the fact that it returns a Future, and that after the Future is completed there is a a new TIFF file per feature.

You probably can base yourself a lot on TestMeteorTFS1Move.